aten/src/ATen/core/TensorMethods.h

```diff
@@ -1276,6 +1280,15 @@ inline bool is_cuda(Tensor self) {
   return self.is_cuda();
 }
 
+inline bool Tensor::is_hip() const {
+  // NB: this is not a native function to avoid dispatching overhead.
+  return impl_->is_hip();
+}
+
+inline bool is_hip(Tensor self) {
```

Comment:
nit: This doesn't take ownership of the Tensor, feel free to pass it as `const Tensor&` to avoid the refcount bump

Reply:
This is from pre-existing code. I'll fix this in a different PR.